### DISCUSSION TITLE
isTableRecordSelectable for admin resource?

### DISCUSSION THREAD

**Discussion by wivaku** [filamentphp/filament#6059] [Help]
For a table I can set [isTableRecordSelectable](https://filamentphp.com/docs/2.x/tables/actions#disabling-record-bulk-actions).

Can I do the same for an admin resource?

Would like to make e.g. only the user rows selectable that don't have a phone.

I tried:

```
    // UserResource

    public function isTableRecordSelectable(): ?Closure
    {
        return fn(User $record): bool => !$record->phone;
    }
```

But this does not seem to work: all records are selectable.

Using something similar on a row level and that works as expected
```
    // UserResource

    public static function table(Table $table): Table
    {
        return $table
            // ...
            ->actions([
                Tables\Actions\Action::make('add Phone')
                    ->visible(fn(User $record): bool => !$record->phone)
                    // ...
```

**Comment by danharrin** [comment]
Put it on the list page class instead of the resource class

**Comment by wivaku** [comment]
Ok thanks @danharrin. It's resource pages that I have. So it is not possible there?

To me it is not always clear what works for both a resource and a list page class, and what works only for either a resource or list page class.

**Reply by danharrin** [reply]
resource pages are pages, so you can customise the heading or subheading per-page.

your IDE will assist on which class to add methods and properties to

**Reply by wivaku** [reply]
Your additional comment in #6061 led to a lightbulb moment. Thanks!
